### ISSUE TITLE
[Talk] Document Intelligence vLLM+k8s

### ISSUE THREAD

**Issue by hedrergudene** [anajsana/pytorch-madrid-group#3]
### 👤 Your name

Antonio Zarauz Moreno

### 💼 Role & affiliation (optional)

AI Technical Staff @Credicorp

### 📌 Talk title

Beyond Document Intelligence: Engineering High-Throughput SLM Document Understanding with Rust, ORT, and vLLM

### 🎬 Format

🎙 Full talk

### 📝 Abstract

Visual Document Understanding (VDU) has transitioned from traditional OCR to generative reasoning, but delivering low-latency inference at scale remains a systems engineering challenge. Processing dense scanned documents requires processing thousands of image tokens per crop without stalling the continuous batching scheduler.

This talk unpacks the architecture behind a scalable, low-latency OCR pipeline built for modern AI agents. We’ll dissect the performance mechanics of combining deterministic layout analysis with generative Small Language Models (SLMs), demonstrating how to pipeline heterogeneous GPU nodes without head-of-line blocking. We will compare traditional Python-based worker architectures against a custom Rust-ORT orchestrator that leverages in-memory zero-copy tensor handoffs and asynchronous dispatching. Finally, we will cover the practical telemetry and parameter tuning required in vLLM to achieve sustained 100% GPU core saturation.

### 🧩 PyTorch ecosystem - what does your talk touch?

- [ ] PyTorch core
- [ ] torch.compile
- [x] vLLM
- [ ] DeepSpeed
- [ ] Unsloth
- [ ] ExecuTorch
- [ ] TorchServe
- [ ] Ray
- [ ] Safetensors
- [x] Kubernetes / Kubeflow
- [ ] Other (mention in abstract)

### 🎯 Short bio

R&D AI director and ML engineer at Credicorp dedicated to transforming research into reliable production systems. With expertise in LLM infrastructure, distributed training, and cloud deployment, he focuses on the practical bridge between prototypes and real-world operations. He brings a deep "builder’s" perspective to Foundational Models training, inference engineering and Kubernetes orchestration.

### 📸 Speaker photo

<img width="1440" height="1440" alt="Image" src="https://github.com/user-attachments/assets/1f656573-ea34-4972-9656-412ccf240e0f" />

### 📅 Availability

The upcoming meetup (October 2026)

### ✅ Confirmation

- [x] I'm ok with the talk being **recorded and shared publicly** (video + slides).

**Comment by oglok** [comment]
Your talk has been accepted to the Pytorch Meetup in Madrid the next 14th of October! 
We'll reach out to you soon for more details!
